agent/bzero/registration/registration.go

```diff
@@ -93,7 +86,75 @@ func Register(log logger.T, apiKey string, envName string, envId string, targetN
 	return response, nil
 }
 
-func post(log logger.T, regInfo BZeroRegRequest, regUrl string) (*http.Response, error) {
+func sendRegisterRequest(log logger.T, regInfo BZeroRegRequest, serviceUrl string) (*http.Response, error) {
+	// Declare our variables
+	var response *http.Response
+
+	// Marshal the regInfo data so we don't do it every time
+	regInfoBytes, err := json.Marshal(regInfo)
+	if err != nil {
+		return response, fmt.Errorf("could not marshal registration request")
+	}
+
+	// Build Registration Endpoint
+	if serviceUrl == "" {
```

Comment:
might want to make sure service url has a scheme so "lucie.bastionzero.com" becomes "https://lucie.bastionzero.com". But this would only be a dev issue and I think we don't need to cater to us

Reply:
zli system tests does always add the scheme because we get it from the config file. So yeah this would only be an issue for devs registering manually.